src/native/eventpipe/ep.c

```diff
@@ -387,7 +409,13 @@ ep_provider_config_init (
 void
 ep_provider_config_fini (EventPipeProviderConfiguration *provider_config)
 {
-	;
+	ep_return_void_if_nok (provider_config != NULL);
+
+	ep_rt_utf8_string_free ((ep_char8_t *)ep_provider_config_get_provider_name (provider_config));
+	ep_rt_utf8_string_free ((ep_char8_t *)ep_provider_config_get_filter_data (provider_config));
```

Comment:
You'll need to review all code paths that were calling ep_provider_config_init to ensure that all of them are handing off ownership of the strings they pass in. For example ep_session_enable_rundown looks like it passes in a global singleton returned by ep_config_get_rundown_provider_name_utf8(). The caller needs to make a copy so that we can safely free it here.